### ISSUE TITLE
support for authorized_dataset in bigquery_dataset_access

### ISSUE THREAD

**Issue by nolabill** [terraform-google-modules/terraform-google-bigquery#141]
### TL;DR

In https://github.com/terraform-google-modules/terraform-google-bigquery/tree/master/modules/authorization, there is a block for authorized_views, I would like to see similar support for authorized_datasets.

### Terraform Resources

```markdown
https://github.com/terraform-google-modules/terraform-google-bigquery/tree/master/modules/authorization
```

### Additional information

Ideally, this is just a stop-gap solution.  In the long-term, I would expect that the bigquery_dataset resource would be able to handle all of this.  Usage of authorized_datasets are the exception and the preference would be that those who are creating a standard BQ dataset would not need to follow the older "access" block method.  However, separating the 2 use-cases, when the details may not always be known at the time of creation, will likely cause problems - meaning we will have to use this method for all BQ resources.
